src/units/unit.hpp

```diff
@@ -1479,6 +1479,13 @@ class unit
 	{
 		return get_abilities(tag_name, loc_);
 	}
+	
+	unit_ability_list get_abilities(const std::string& tag_name, const map_location& loc, const_attack_ptr weapon) const;
```

Comment:
The weapon parameter should be optional and default to `nullptr` so that the function pair with no weapon can then be removed.